### PR TITLE
Added workflow for continous integeration

### LINKED ISSUE #48: Move continuous deployment to GitHub Actions
**Is your feature request related to a problem? Please describe.**
We currently use TravisCI. We'd like to use GitHub Actions instead.

**Describe the solution you'd like**
The project should build on GitHub Actions and the automatic build and deployment should work.

### PR DESCRIPTION
**Pull request by techievivek**
Fixes #48 

## Description
Added the code for continuous integeration when any push event occurs.

## Other information
For now I have added the syntax for CI and with the help of admin I will update for the CD too.

## Checklist
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository.
- [x] My commit messages follow [best practices][best_practices].
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin
```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

### CODE REVIEW

**Comment by techievivek** [comment]
@kgodey Please have a look and also help me how to I add the syntax for CD.

**Comment by techievivek** [comment]
@kgodey I have added both CI/CD part using github actions and also tested for this repo and a [sample repo](https://github.com/techievivek/lektor-deployment) of mine(built using lektor), and its working. Please have a look and let me know if I can improve upon anything.
Thank you

File: .github/workflows/lektor-server.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+     run: |
+      pip install pipenv
+      pipenv install
+      pipenv run lektor build -f webpack --output-path ./build 
+   - name: Deploy
+     uses: peaceiris/actions-gh-pages@v3
+     with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          publish_dir: ./build
+          publish_branch: gh-pages
```
Comment (kgodey) [inline .github/workflows/lektor-server.yml]: We deploy to a different repo (https://github.com/creativecommons/creativecommons.github.io), we do not use a `gh-pages` branch.

Comment (kgodey) [reply .github/workflows/lektor-server.yml]: Please use the `lektor deploy ghpages` script as shown in the current Travis config: https://github.com/creativecommons/creativecommons.github.io-source/blob/master/.travis.yml

File: .github/workflows/lektor-server.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+name: Building and Deploying to CC opensource site
```
Comment (kgodey) [inline .github/workflows/lektor-server.yml]: Please rename this to "Build and Deploy CC Open Source"

File: .github/workflows/lektor-server.yml (hunk 1 above)
Comment (techievivek) [reply .github/workflows/lektor-server.yml]: Will running the command `lektor deploy ghpages` do all the work? 
I am not sure how I will go about deploying to a different repo? Are there any other commands I have to run along?

Comment (kgodey) [reply .github/workflows/lektor-server.yml]: You don't need to run anything else. The deployment is configured here: https://github.com/creativecommons/creativecommons.github.io-source/blob/master/.lektorproject

**Comment by kgodey** [comment]
One more thing – could you rewrite your commits so that they are more descriptive? `Update lektor-server.yml` doesn't really say anything about the work you did.

It may be easiest to do a `git reset` to the commit before your work and then create a single new commit that's more descriptive.

**Comment by techievivek** [comment]
Yes, I have gone through the above file but didn't understand much.
Can I close this PR and start and new one? Or else I will do a reset from very first.

**Comment by kgodey** [comment]
Sure, go ahead and close this one.

**Pull request closed without merging**